src/fdc3/dotnet/DesktopAgent/src/MorganStanley.ComposeUI.DesktopAgent/Fdc3DesktopAgent.cs

```diff
@@ -901,28 +837,54 @@ public async ValueTask<RaiseIntentResult<RaiseIntentResponse>> RaiseIntentForCon
             raiseIntentSpecification.TargetAppMetadata = appIntent.Apps.ElementAt(0);
             return await RaiseIntentToApplication(raiseIntentSpecification);
         }
-        
-        var resolverUIResult = await WaitForResolverUIAsync(appIntent.Apps);
+
+        var resolverUIResult = await _resolverUI.SendResolverUIRequest(appIntent.Apps);
         if (resolverUIResult != null && resolverUIResult.Error == null)
         {
             raiseIntentSpecification.TargetAppMetadata = (AppMetadata) resolverUIResult.AppMetadata!;
             return await RaiseIntentToApplication(raiseIntentSpecification);
         }
-        
+
         if (resolverUIResult?.Error != null)
         {
             return new()
             {
                 Response = RaiseIntentResponse.Failure(resolverUIResult.Error)
             };
         }
-        
+
         return new()
         {
             Response = RaiseIntentResponse.Failure(ResolveError.UserCancelledResolution)
         };
     }
 
+    private IEnumerable<AppIntent> FilterAppIntentsByAppId(IEnumerable<AppIntent> source, AppIdentifier appId)
+    {
+        // Semantically we know this is not null or empty
+        foreach (var intent in source)
+        {
+            foreach (var app in intent.Apps)
+            {
+                var matches = true;
+                if (appId.AppId != null && appId.AppId != app.AppId)
```

Comment:
AppId property won't be/shouldn't be null if the AppIdentifier is not null. is it necessary to check against null though?

Reply:
Semantically the intent is "if the request contains a non-null value, filter on it" - it is true though that if the filter contains null, it will never match, so I can remove the null check

Reply:
The point is if the filter is null, it shouldn't check for equivalency, so yes, the null check is important